### PR TITLE
[BEAM-1663] Use stable naming strategy for ByteBuddy invokers

### PR DESCRIPTION
**Pull request by kennknowles**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`.
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

This helps to coalesce related failures across multiple JVM instances, over time, across machines, etc.

### CODE REVIEW

File: sdks/java/core/pom.xml [hunk 1]
```diff
@@ -315,6 +315,19 @@
       <scope>test</scope>
     </dependency>
 
+    <!-- 
+    bytebuddy has provided scope dep on SuppressFBWarnings 
+    annotations (to avoid a real dep) and these are
+    retained on certain .class files, necessitating
+    this provided dependency to instantiate those classes
+    -->
+    <dependency>
```
Comment (kennknowles) [inline sdks/java/core/pom.xml]: @davorbonaci can you clarify Apache licensing here? The root cause of our needing something here is a similar construct in the bytebuddy pom: https://github.com/raphw/byte-buddy/blob/master/pom.xml#L163

**Comment by kennknowles** [comment]
run flink validatesrunner

**Comment by kennknowles** [comment]
run dataflow validatesrunner

**Comment by kennknowles** [comment]
We've been seeing some flakes in the Flink runner `ValidatesRunner` tests, especially around side inputs. I believe this result is one such flake. Re-running anyhow.

**Comment by kennknowles** [comment]
run flink validatesrunner

**Comment by kennknowles** [comment]
R: @bjchambers more bytebuddy fun

File: sdks/java/core/pom.xml (hunk 1 above)
Comment (kennknowles) [inline sdks/java/core/pom.xml]: Cancel that - we just won't test at this fine granularity. A change-detector test suffices.

File: sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/reflect/OnTimerInvokersTest.java [hunk 2]
```diff
@@ -105,4 +105,36 @@ public void onMyTimer(BoundedWindow window) {
...
+  @Test
+  public void testStableName() {
+    OnTimerInvoker<Void, Void> invoker =
+        OnTimerInvokers.forTimer(new StableNameTestDoFn(), StableNameTestDoFn.TIMER_ID);
+
+    assertThat(
+        invoker.getClass().getName(),
+        equalTo(
+            String.format(
+                "%s$%s$%s$dGltZXItaWQud2l0aCBzcGVjaWFsQ2hhcnN7fQ",
```
Comment (bjchambers) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/reflect/OnTimerInvokersTest.java]: This should be "%s$%s$%s$%s", or the last argument is ignored (inlined here anyways).

Comment (kennknowles) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/reflect/OnTimerInvokersTest.java]: Done. Trailed off mid-thought it seems. Thanks for the catch.

**Pull request merged**